spec/presenters/submission_presenter_spec.rb

```diff
@@ -1,68 +1,89 @@
+      is_expected.to eq(submission)
+      is_expected.not_to be_a(Submission)
     end
   end
 
   describe "#average_rate" do
     describe "when submission is unrated" do
-      let!(:submission) { FactoryGirl.build(:submission) }
+      let(:rates) { double }
+      subject { submission_presenter.average_rate }
 
-      it { expect(subject.average_rate).to eq(nil) }
-    end
+      before { allow(submission).to receive(:rated?).and_return(false) }
 
-    describe "when submission is rated and has no rates" do
-      before do
-        allow(Setting).to receive(:get).and_return(
-          FactoryGirl.create(:setting, required_rates_num: 0))
-      end
+      it { is_expected.to eq(nil) }
+    end
 
-      let!(:submission) do
-        FactoryGirl.build(:submission, :rated, required_rates_num: 0)
-      end
+    describe "when submission is rated" do
+      # overriden with real object since average is calculated in the db
+      let(:submission) { FactoryGirl.create(:submission, :rated, rate_value: 2) }
+      let(:rates) { submission.rates }
+      subject { submission_presenter.average_rate }
 
-      it { expect(subject.average_rate).to eq(0) }
+      it { is_expected.to eq(2) }
     end
+  end
+
+  describe "delegates methods to submission_repository" do
+    describe "#next_to_rate" do
+      let(:rates) { double }
+      let(:result) { double }
+      subject { submission_presenter.next_to_rate }
 
-    describe "when submission is rated and has some rates" do
       before do
-        allow(Setting).to receive(:get).and_return(
-          FactoryGirl.create(:setting, required_rates_num: 3))
+        date = double
+        allow(submission).to receive(:created_at).and_return(date)
+        allow(submission_repository).to receive(:next_to_rate).with(date).and_return(result)
       end
 
-      let!(:submission) do
-        FactoryGirl.build(:submission, :rated, { required_rates_num: 3, rate_value: 2 })
+      it { is_expected.to eq(result) }
+    end
+
+    describe "#previous_to_rate" do
+      let(:rates) { double }
+      let(:result) { double }
+      subject { submission_presenter.previous_to_rate }
+
+      before do
+        date = double
+        allow(submission).to receive(:created_at).and_return(date)
+        allow(submission_repository).to receive(:previous_to_rate).with(date).and_return(result)
       end
 
-      it { expect(subject.average_rate).to eq(2) }
+      it { is_expected.to eq(result) }
     end
   end
 
-  describe "delegates methods to submissions_repository" do
-    let!(:submission) { FactoryGirl.create(:submission) }
+  describe "#rates_count" do
+    let(:rates) { double }
+    subject { submission_presenter.rates_count }
 
-    context "#next_to_rate" do
-      it do
-        expect(submissions_repository).to receive(:next_to_rate).with(submission.created_at)
-        subject.next_to_rate
-      end
-    end
+    before { allow(rates).to receive(:count).and_return(1) }
```

Comment:
Here I think it's not that bad, you can leave it. Just have in mind for future to be careful with stubbing, to not stub implementation too much and not stub private methods